app/src/main/java/com/nilhcem/blenamebadge/adapter/DrawableAdapter.kt

```diff
@@ -24,55 +21,29 @@ class DrawableAdapter(private val context: Context?, private val list: List<Draw
     override fun getItemId(position: Int) = position.toLong()
 
     override fun onBindViewHolder(holder: DrawableItemHolder, position: Int) {
-        holder.bind(list[position])
+        holder.apply {
+            bind(drawableList[position], selectedPosition, position)
+            listener = onDrawableSelected
+        }
     }
 
-    fun getSelectedItem(): DrawableInfo? {
-        return if (selectedPosition == -1) null else list[selectedPosition]
+    fun addAll(newDrawableList: List<DrawableInfo>) {
+        if (drawableList.isNotEmpty()) drawableList.clear()
+        drawableList.addAll(newDrawableList)
+        notifyDataSetChanged()
     }
 
-    fun setListener(listener: OnDrawableSelected) {
-        this.listener = listener
+    fun setSelectedDrawablePosition(position: Int) {
+        selectedPosition = if (selectedPosition == position) -1 else position
     }
 
-    override fun getItemCount() = list.size
-
-    inner class DrawableItemHolder(itemView: View) : RecyclerView.ViewHolder(itemView) {
-
-        private val card: LinearLayout = itemView.findViewById(R.id.card)
-        private val image: ImageView = itemView.findViewById(R.id.image)
-
-        init {
-            card.setOnClickListener {
-                changeCardBackgrounds()
-                listener?.onSelected(getSelectedItem())
-            }
-        }
-
-        fun bind(drawableInfo: DrawableInfo) {
-            image.setImageDrawable(drawableInfo.image)
-
-            card.background = when {
```

Comment:
Adapter is the place where every position related code need to be present. Don't shift the color change and other stuff to fragments or activities. I don't think it is a good practise

Reply:
I didn't move color change or view change to the fragment or activities. They are in the ViewHolder classes. And the ViewHolder classes are moved from the Adapter classes because I think separating those classes would make the readability of the code better

Reply:
@anhanh11001, my bad, I did not notice that you shifted it to a new file. Definitely it will be more readable.